SA3/qa_web/views.py

```diff
@@ -76,11 +77,29 @@ def questions(request):
             return render(request, 'qa_web/questionspage.html', context={})
 
 
-def answers(request, id):
-        q = get_object_or_404(Questions, pk=id)
-        return render(request, 'qa_web/answerspage.html', {'currentQuestion': q})
+def answers(request, id_):
+    q = get_object_or_404(Questions, pk=id_)
+    if request.method == 'POST' and 'answer_form' in request.POST: #Update's database when somebody answers a question
+        form = AnswersForm(request.POST)
+        if form.is_valid(): 
+            Answers.objects.create(content=request.POST['content'], owner=request.user, question=q)
+    elif request.method == 'POST' and 'deselect' in request.POST:  #Update's database when somebody deselects best answer.
+        updateAnswer = Answers.objects.filter(question=q, correct_answer=True).last()
+        updateAnswer.correct_answer = False;
+        updateAnswer.save();
+    else: #Update's database when somebody selects a best answer.
+        q_answers = Answers.objects.filter(question=q, correct_answer=False)
```

Comment:
Instead of iterating through all of a question's answers and compare it to the id found in the POST dictionary, you should retrieve the id from the dictionary's keys and retrieve the answer that way. Furthermore, this condition should rather be `elif request.method == 'POST'` since otherwise GET requests will execute the following code unnecessarily.  Here's how I would retrieve the answer that needs to be modified.
```
# This filters through the POST keys and parses the id from keys formatted like 'select_<id>'
answer_id = [int(key.replace('select_', '')) for key in request.POST.keys() if key.startswith('select_')]
if answer_id:
    updateAnswer = Answers.objects.get(pk = answer_id[0])
    [...]
```